iris_grib/tests/unit/save_rules/test_product_definition_template_15.py

```diff
@@ -0,0 +1,161 @@
+# (C) British Crown Copyright 2018, Met Office
+#
+# This file is part of iris-grib.
+#
+# iris-grib is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# iris-grib is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with iris-grib.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Unit tests for :func:`iris_grib._save_rules.product_definition_template_15`
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris_grib.tests first so that some things can be initialised before
+# importing anything else.
+import iris_grib.tests as tests
+
+from cf_units import Unit
+import gribapi
+import mock
+
+from iris.coords import CellMethod, DimCoord
+import iris.tests.stock as stock
+
+from iris_grib._save_rules import product_definition_template_15
+
+
+class TestSpatialProcessingIdentifiers(tests.IrisGribTest):
+    def setUp(self):
+        self.cube = stock.lat_lon_cube()
+        # Add scalar time coord so that product_definition_template_common
+        # doesn't get upset.
+        t_coord = DimCoord([424854.], standard_name='time',
+                           units=Unit('hours since 1970-01-01 00:00:00',
+                                      calendar='gregorian'))
+        self.cube.add_aux_coord(t_coord)
+        # Rename cube to avoid warning about unknown discipline/parameter.
+        self.cube.rename('WAFC_CAT_potential')
+
+    @mock.patch.object(gribapi, 'grib_set')
+    def test_cell_method(self, mock_set):
+        cube_0 = self.cube
+        cube_0.attributes = dict(spatial_processing_type=0)
+        cell_method = CellMethod(method='mean', coords=['area'])
+        cube_0.add_cell_method(cell_method)
+
+        # If the cube has a cell method attached then it should not have any
+        # interpolation on the data, so spatial processing code should be 0 and
+        # number of points used should be 0.
+        product_definition_template_15(cube_0, mock.sentinel.grib)
+        mock_set.assert_any_call(mock.sentinel.grib,
+                                 "productDefinitionTemplateNumber", 15)
+        mock_set.assert_any_call(mock.sentinel.grib,
+                                 "spatialProcessing", 0)
+        mock_set.assert_any_call(mock.sentinel.grib,
+                                 "numberOfPointsUsed", 0)
```

Comment:
Test statistical type.

Reply:
Actually, changes introduced regarding the lookup dictionary mean that we either save `numberOfPoints` or `statisticalProcess` depending on whether there is a cell method or not.

We can therefore only test for `statisticalProcess` when there is a cell method present, and when there is no cell method then we can only test for `numberOfPoints`.